### ISSUE TITLE
Document the biggest issues in code health

### ISSUE THREAD

**Issue by meiqimichelle** [18F/testing-grouplet#38]
There are a 'top few' things that services like code climate test for. We should document those in this repo so that we can start to come up with actionable recommendations for how to fix them.

I'm not exactly an expert in this sort of thing, but [this website](http://jscomplexity.org/complexity) seems to have a good summary of complexity metrics and a great overview of what matters, what doesn't, and how to use the info. It lists:
- lines of code
- number of parameters
- cyclomatic complexity
- cyclomatic complexity density
- Halstead complexity measures
- maintainability index
- dependencies
- first-order density
- change cost
- core size

Code climate, on the other hand, has these categories for categories of things:
<img width="249" alt="screenshot 2016-02-05 00 06 08" src="https://cloud.githubusercontent.com/assets/4827522/12840023/5a03c704-cb9c-11e5-9b3d-1dc5b61a150e.png">

Is this the list that matters? What else should be here?

My goal is to start to figure out what we should be addressing, and then while we work on the dashboard design, other folks can be writing guides or interactive teachings like Mike's unit testing thing that will be the instructions on how to fix the problems. Maybe it's even just links to articles on what to do, Who knows?

/cc @mbland @monfresh @jmcarp

**Comment by mbland** [comment]
Yes, the hybrid approach is what I'm suggesting: Giving people both the power (tools) and the knowledge to do the right thing, so one day the right thing becomes the easiest thing to do. I've also written a great deal myself about how testing addresses many of these. [Goto Fail, Heartbleed, and Unit Testing Culture](http://martinfowler.com/articles/testing-culture.html) being the most comprehensive, probably.
